ipaserver/install/httpinstance.py

```diff
@@ -570,3 +576,16 @@ def start_tracking_certificates(self):
         db = certs.CertDB(self.realm, nssdir=paths.HTTPD_ALIAS_DIR)
         db.track_server_cert(self.cert_nickname, self.principal,
                              db.passwd_fname, 'restart_httpd')
+
+    def request_service_keytab(self):
+        super(HTTPInstance, self).request_service_keytab()
+
+        if self.master_fqdn is not None:
+            service_dn = DN(('krbprincipalname', self.principal),
+                            api.env.container_service,
+                            self.suffix)
+
+            ldap_uri = ipaldap.get_ldap_uri(self.master_fqdn)
+            with ipaldap.LDAPClient(ldap_uri) as remote_ldap:
+                remote_ldap.simple_bind(ipaldap.DIRMAN_DN, self.dm_password)
+                replication.wait_for_entry(remote_ldap, service_dn, timeout=60)
```

Comment:
I am afraid this will not work in DL1 since directory manager password is randomly generated at the start of the replica and you cannot use it to connect to remote master (you will get ACIError) until the very end of replica install when Custodia pulls in the 'real' DM password from master.

IIRC http installer has `self.promote` attribute set during replica install, so you can use the good ole ugly promote hack:

```python
            with ipaldap.LDAPClient(ldap_uri) as remote_ldap:
                if self.promote:
                    remote_ldap.gssapi_bind()
                else:
                    remote_ldap.simple_bind(ipaldap.DIRMAN_DN, self.dm_password)
```

That should keep us covered for all cases (DL0, DL1, migrations). If you feel like doing a little bit of cleanup today then you can factor this part out to a separate method of parent class that returns the connection to master, but I do not insist on this improvement.